test/singletons.test.js

```diff
@@ -222,22 +223,54 @@ describe('singletons', function() {
   it('resolving singleton before first sync, then signing in, should result in correct number of instances', async function () {
     await this.registerUser();
     /** Create prefs and associate them with account */
-    const ogPrefs = await insertPrefsPayload(this.application);
+    const ogPrefs = await findOrCreatePrefsSingleton(this.application);
     await this.application.sync(syncOptions);
     this.application = await Factory.signOutApplicationAndReturnNew(this.application);
 
     /** Create another instance while signed out */
-    await insertPrefsPayload(this.application);
+    await findOrCreatePrefsSingleton(this.application);
     await Factory.loginToApplication({
       application: this.application,
       email: this.email,
       password: this.password
     });
     /** After signing in, the instance retrieved from the server should be the one kept */
-    const latestPrefs = await insertPrefsPayload(this.application);
+    const latestPrefs = await findOrCreatePrefsSingleton(this.application);
     expect(latestPrefs.uuid).to.equal(ogPrefs.uuid);
-    const allPrivs = this.application.itemManager.nonErroredItemsForContentType(ogPrefs.content_type);
-    expect(allPrivs.length).to.equal(1);
+    const allPrefs = this.application.itemManager.nonErroredItemsForContentType(ogPrefs.content_type);
+    expect(allPrefs.length).to.equal(1);
+  });
+
+  it('keeps local singleton when it is created before sync', async function () {
```

Comment:
I'm really not sure what this test proves anymore. I think here you seem to be just testing storage behavior rather than anything to do with singletons: if you delete something from storage and replace it with another entity, then surely when the app initializes, it will only have the new one, as it's like old one never even existed if you surreptitiously delete it from storage.

Can you describe what you're trying to test? How is `keeps local singleton when it is created before sync` describing the patch you made?

Reply:
It's not a test that is related to the PR anymore, but rather one that proves that a server singleton does not have priority outside of downloadFirstSync. That way if this behavior ever changes the test will fail, prompting us to question whether this was intended or not.

You're right that I do need to write another test for this specific PR, though 😅

Reply:
If you want to prove that an incoming server item will not override the local copy, then you need to clear the sync token and sync again from scratch. Otherwise I'm not sure this test proves anything..